sql/hive/src/main/scala/org/apache/spark/sql/hive/test/TestHive.scala

```diff
@@ -139,12 +139,9 @@ private[hive] class TestHiveSparkSession(
 
   assume(sc.conf.get(CATALOG_IMPLEMENTATION) == "hive")
 
-  // TODO: Let's remove HiveSharedState and TestHiveSessionState. Otherwise,
-  // we are not really testing the reflection logic based on the setting of
-  // CATALOG_IMPLEMENTATION.
   @transient
-  override lazy val sharedState: HiveSharedState = {
-    existingSharedState.getOrElse(new HiveSharedState(sc))
+  override lazy val sharedState: SharedState = {
```

Comment:
`sharedState` in `SparkSession` is unable to access the `existingSharedState` defined in `TestHiveSparkSession`.  We are unable to override it because [it is private](https://github.com/apache/spark/blob/18c2c92580bdc27aa5129d9e7abda418a3633ea6/sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala#L55).